### PR TITLE
BAHIR-130

### PR DESCRIPTION
**Pull request by romeokienzler**
In case number of requests per second are limited; just wait for a second and try again

### CODE REVIEW

**Comment by ApacheBahir** [comment]
Can one of the admins verify this patch?

File: sql-cloudant/src/main/scala/org/apache/bahir/cloudant/common/JsonStoreDataAccess.scala [hunk 1]
```diff
@@ -125,6 +125,19 @@ class JsonStoreDataAccess (config: CloudantConfig)  {
     val clRequest: HttpRequest = getClRequest(url, postData)
 
     val clResponse: HttpResponse[String] = clRequest.execute()
+          
+    //BAHIR-130 start
+    //In cloudant Lite number of requests pre second are limited.
+    //Just wait and try again
+    clResponse.code match {
+        case 429 => {
+            Thread sleep 1000
```
Comment (emlaver) [inline sql-cloudant/src/main/scala/org/apache/bahir/cloudant/common/JsonStoreDataAccess.scala:134]: If the server is busy, we don't want this to loop and keep trying forever.  We should have a set number of retries (our other Cloudant java library uses 3 retries) and exit with a 429 error if we reach that limit.

Comment (emlaver) [reply sql-cloudant/src/main/scala/org/apache/bahir/cloudant/common/JsonStoreDataAccess.scala:134]: For each retry we should consider increasing the wait 'backoff' time.  Use something like `long sleepTime = initialSleep * Math.round(Math.pow(2, attempt))` to double the sleep time based on the number of retry attempts.  Our other libraries start with an initial sleep time of 250 ms.  Here's the [retry method](https://github.com/cloudant/java-cloudant/blob/515cf156751cd67d714a206d1bc39a5489f8c156/cloudant-http/src/main/java/com/cloudant/http/interceptors/Replay429Interceptor.java#L83.) in our java library that may be helpful.

Comment (emlaver) [reply sql-cloudant/src/main/scala/org/apache/bahir/cloudant/common/JsonStoreDataAccess.scala:134]: Also, would be good to have logging for:
- When there's too many requests and attempting a retry with x seconds
- Any problems/interrupt during the 'backoff' wait time

**Comment by ckadner** [comment]
@romeokienzler -- are you still working on incorporating @emlaver review comments?

**Comment by ckadner** [comment]
@romeokienzler -- any updates or progress?

**Comment by romeokienzler** [comment]
Sorry for the late reply. I'm using my patch in productions since a couple of weeks with great success. So I'll update this thread and also JIRA by incorporating my findings so that we can agree how to proceed.

**Comment by romeokienzler** [comment]
Will be fixed through migration to java-cloudant (see comments in [JIRA](https://issues.apache.org/jira/browse/BAHIR-130) for details)

**Pull request closed without merging**